pyem/metadata.py

```diff
@@ -292,23 +292,80 @@ def cryosparc_2_cs_ctf_parameters(cs, df=None):
     log = logging.getLogger('root')
     if df is None:
         df = pd.DataFrame()
-    if 'ctf/tilt_A' in cs.dtype.names:
-        log.debug("Recovering beam tilt and converting to mrad")
-        df[star.Relion.BEAMTILTX] = np.arcsin(cs['ctf/tilt_A'][:, 0] / cs['ctf/cs_mm'] * 1e-7) * 1e3
-        df[star.Relion.BEAMTILTY] = np.arcsin(cs['ctf/tilt_A'][:, 1] / cs['ctf/cs_mm'] * 1e-7) * 1e3
-    if 'ctf/shift_A' in cs.dtype.names:
-        pass
-    if 'ctf/trefoil_A' in cs.dtype.names:
-        pass
-        # df[star.Relion.ODDZERNIKE] = cs['ctf/trefoil_A']
-    if 'ctf/tetrafoil_A' in cs.dtype.names:
-        pass
-        # df[star.Relion.EVENZERNIKE] = cs['ctf/tetra_A']
+    if 'ctf/accel_kv' in cs.dtype.names:
+        # Calculate electron wavelentgh
+        # https://www.jeol.co.jp/en/words/emterms/search_result.html?keyword=wavelength%20of%20electron
+        E = cs['ctf/accel_kv']  # convert units?
+        wavelength = 1.23e3 / np.sqrt(E*(1 + 9.78e-7 * E))  # ? TODO: check this
```

Comment:
Should be checked for correctness.
https://www.jeol.co.jp/en/words/emterms/search_result.html?keyword=wavelength%20of%20electron

Reply:
Relion uses `lamb = 12.2643247 / np.sqrt(kv * (1. + kv * 0.978466e-6))` which is about the same but with more precision. I think this ends up being numerically sensitive and the precision is probably warranted.

Reply:
That's a good point about the extra precision. I've updated the code, thanks!